tests/dvslib/dvs_database.py

```diff
@@ -0,0 +1,397 @@
+"""
+    dvs_database contains utilities for interacting with redis when writing
+    tests for the virtual switch.
+"""
+from __future__ import print_function
+
+import time
+import collections
+import pytest
+
+from swsscommon import swsscommon
+
+APP_DB_ID = 0
+ASIC_DB_ID = 1
+COUNTERS_DB_ID = 2
+CONFIG_DB_ID = 4
+FLEX_COUNTER_DB_ID = 5
+STATE_DB_ID = 6
+
+
+# PollingConfig provides parameters that are used to control polling behavior
+# when accessing redis:
+# - polling_interval: how often to check for updates in redis
+# - timeout: the max amount of time to wait for updates in redis
+# - strict: if the strict flag is set, failure to receive updates will cause
+#           the polling method to cause tests to fail (e.g. assert False)
+PollingConfig = collections.namedtuple('PollingConfig', 'polling_interval timeout strict')
+
+
+class DVSDatabase(object):
+    """
+        DVSDatabase provides access to redis databases on the virtual switch.
+
+        By default, database operations are configured to use
+        `DEFAULT_POLLING_CONFIG`. Users can specify their own PollingConfig,
+        but this shouldn't typically be necessary.
+    """
+    DEFAULT_POLLING_CONFIG = PollingConfig(polling_interval=0.01, timeout=5, strict=True)
+
+    def __init__(self, db_id, connector):
+        """
+            Initializes a DVSDatabase instance.
+
+            Args:
+                db_id (int): The integer ID used to identify the given database
+                    instance in redis.
+                connector (str): The I/O connection used to communicate with
+                    redis (e.g. unix socket, tcp socket, etc.).
+
+            NOTE: Currently it's most convenient to let the user specify the
+            connector since it's set up in the dvs fixture. We may abstract
+            this further in the future as we refactor dvs.
+        """
+
+        self.db_connection = swsscommon.DBConnector(db_id, connector, 0)
+
+    def create_entry(self, table_name, key, entry):
+        """
+            Adds the mapping {`key` -> `entry`} to the specified table.
+
+            Args:
+                table_name (str): The name of the table to add the entry to.
+                key (str): The key that maps to the entry.
+                entry (Dict[str, str]): A set of key-value pairs to be stored.
+        """
+
+        table = swsscommon.Table(self.db_connection, table_name)
+        formatted_entry = swsscommon.FieldValuePairs(entry.items())
+        table.set(key, formatted_entry)
+
+    def get_entry(self, table_name, key,
+                  polling_config=DEFAULT_POLLING_CONFIG):
```

Comment:
This function is doing wait_for_entry(), maybe you should name the function accordingly?

This comment is applicable to other polling functions too.

Is there a way to disable polling and just do a one shot check? If that is the case, then maybe name the function like wait_entry_until()?

Reply:
> Is there a way to disable polling and just do a one shot check? If that is the case, then maybe name the function like wait_entry_until()?

You could do that pretty easily by zeroing all the polling params and turning off strict checking. I could make a default for that (like `DISABLE_POLLING_CONFIG`), or even provide a set of wrappers for one-shot checking, I'm just not sure how useful that will be.